devtools/maven/src/main/java/io/quarkus/maven/GenerateExtensionsJsonMojo.java

```diff
@@ -143,56 +186,90 @@ public void execute() throws MojoExecutionException, MojoFailureException {
     }
 
-    private void processDependency(Artifact artifact, JsonArrayBuilder extJsonBuilder) throws IOException {
+    private JsonObject processDependency(Artifact artifact) throws IOException {
         final Path path = artifact.getFile().toPath();
         if (Files.isDirectory(path)) {
-            processMetaInfDir(artifact, path.resolve(BootstrapConstants.META_INF), extJsonBuilder);
+            return processMetaInfDir(artifact, path.resolve(BootstrapConstants.META_INF));
         } else {
             try (FileSystem artifactFs = ZipUtils.newFileSystem(path)) {
-                processMetaInfDir(artifact, artifactFs.getPath(BootstrapConstants.META_INF), extJsonBuilder);
+                return processMetaInfDir(artifact, artifactFs.getPath(BootstrapConstants.META_INF));
             }
         }
     }
 
-    private boolean processMetaInfDir(Artifact artifact, Path metaInfDir, JsonArrayBuilder extJsonBuilder)
+    private JsonObject processMetaInfDir(Artifact artifact, Path metaInfDir)
             throws IOException {
         if (!Files.exists(metaInfDir)) {
-            return false;
+            return null;
         }
         final Path p = metaInfDir.resolve(BootstrapConstants.EXTENSION_PROPS_JSON_FILE_NAME);
         if (!Files.exists(p)) {
             final Path props = metaInfDir.resolve(BootstrapConstants.DESCRIPTOR_FILE_NAME);
             if (Files.exists(props)) {
-                extJsonBuilder.add(Json.createObjectBuilder()
+                return Json.createObjectBuilder()
                         .add("artifactId", artifact.getArtifactId())
                         .add("groupId", artifact.getGroupId())
                         .add("version", artifact.getVersion())
                         .add("name", artifact.getArtifactId())
-                        .build());
+                        .build();
             }
-            return false;
+            return null;
         }
-        processPlatformArtifact(artifact, p, extJsonBuilder);
-        return true;
+        return processPlatformArtifact(artifact, p);
     }
 
-    private void processPlatformArtifact(Artifact artifact, Path descriptor, JsonArrayBuilder extJsonBuilder)
+    private JsonObject processPlatformArtifact(Artifact artifact, Path descriptor)
             throws IOException {
         try (InputStream is = Files.newInputStream(descriptor)) {
             try (JsonReader reader = Json.createReader(is)) {
                 final JsonObject object = reader.readObject();
                 debug("Adding Quarkus extension %s:%s", object.get("groupId"), object.get("artifactId"));
-                extJsonBuilder.add(object);
+                return object;
             }
         } catch (IOException e) {
             throw new IOException("Failed to parse " + descriptor, e);
         }
     }
 
+    private String extensionId(JsonObject extObject) {
+        return extObject.getString("groupId", "") + "." + extObject.getString("artifactId");
```

Comment:
Why not keep the traditional `:` as a separator? It'll work either way but it'l be more intuitive and easier to read in case we need to display those if it is `:`.

Reply:
Done.